File: java-plugin-netlogo-maven/src/main/java/org/test/MyMessage.java
package org.test;

import org.nlogo.api.*;

public class MyMessage extends DefaultReporter {

    public Syntax getSyntax() {
        return Syntax.reporterSyntax(Syntax.StringType());
    }

    public Object report(Argument args[], Context context)  {
        return "hello world";
    }
}

File: scala-plugin-netlogo-sbt/src/MyMessage.scala
package org.test

import org.nlogo.api
import org.nlogo.api._

class MyMessage extends api.DefaultReporter {
  override def getSyntax: Syntax = {
    return Syntax.reporterSyntax(Syntax.StringType)
  }

  def report(args: Array[api.Argument], context: api.Context):AnyRef = {
    return "hello world"
  }
}
